nixos/modules/profiles/base.nix

```diff
@@ -51,7 +51,9 @@
   ];
 
   # Include support for various filesystems.
-  boot.supportedFilesystems = [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "zfs" "ntfs" "cifs" ];
+  boot.supportedFilesystems = [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" ] ++
+    lib.optionals (lib.meta.availableOn pkgs.stdenv.hostPlatform pkgs.zfs) [ "zfs" ] ++
+    [ "ntfs" "cifs" ];
```

Comment:
Let's reorder this so we don't have the last two unnecessarily in a different list.  It shouldn't cause many rebuilds.

You could also use `lib.optional` if you wanted.

Reply:
No problem.  I wasn't sure if the order of the filesystems in that list was significant.  529e6689510ff9aa803b5cb2592715b664cc7632